campaignion_tracking/campaignion_tracking.module

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * @file
+ * Hook implementations for the Campaignion Tracking module.
+ */
+
+use Drupal\campaignion_action\Loader;
+use Drupal\little_helpers\Webform\Submission;
+use Drupal\little_helpers\Webform\Webform;
+
+/**
+ * Implements hook_preprocess_html().
+ */
+function campaignion_tracking_preprocess_html(&$vars) {
+  $module_path = drupal_get_path('module', 'campaignion_tracking');
+
+  // Add javascript for common tracking structure.
+  drupal_add_js($module_path . '/js/campaignion_tracking_common.min.js', [
+    'scope' => 'header',
+    'group' => JS_LIBRARY,
+    'preprocess' => FALSE,
+  ]);
+  // Add javascript which triggers messages to the tracker.
+  drupal_add_js($module_path . '/js/campaignion_tracking.js');
+}
+
+/**
+ * Implements hook_node_view().
+ */
+function campaignion_tracking_node_view($node, $view_mode, $langcode) {
+  $type = Loader::instance()->type($node->type);
+  $is_donation = $type && $type->isDonation();
+
+  // General information about the node.
+  drupal_add_js([
```

Comment:
This should use: `$node->content['#attached']` (or something similar).

Reply:
changed